plugins/modules/omevv_baseline_profile.py

```diff
@@ -0,0 +1,589 @@
+            self.module.exit_json(msg=CHANGES_NOT_FOUND_MSG, changed=False)
+
+        if self.module.check_mode and self.module._diff:
+            self.module.exit_json(msg=CHANGES_FOUND_MSG, diff=diff, changed=True)
+
+        if self.module.check_mode:
+            self.module.exit_json(msg=CHANGES_FOUND_MSG, changed=True)
+
+        self.perform_modify_baseline_profile(new_payload, diff)
+
+
+class DeleteBaselineProfile(BaselineProfile):
+
+    def __init__(self, module, rest_obj, profile_name):
+        super().__init__(module, rest_obj)
+        self.profile_name = profile_name
+
+    def diff_mode_check(self, payload):
+        diff = {}
+        diff_dict = {}
+        diff_dict["name"] = payload["name"]
+        diff_dict["description"] = payload["description"]
+        diff_dict["firmwareRepoId"] = payload["firmwareRepoId"]
+        diff_dict["firmwareRepoName"] = payload["firmwareRepoName"]
+        diff_dict["clusterGroups"] = payload["clusterGroups"]
+        if self.module._diff:
+            diff = dict(
+                before=diff_dict,
+                after={}
+            )
+        return diff
+
+    def perform_delete_baseline_profile(self, profile_resp):
+        diff = {}
+        diff = self.diff_mode_check(profile_resp)
+        vcenter_uuid = self.module.params.get('vcenter_uuid')
+        resp = self.omevv_baseline_obj.delete_baseline_profile(profile_resp["id"], vcenter_uuid)
+        if resp.success:
+            if self.module._diff:
+                self.module.exit_json(msg=SUCCESS_DELETION_MSG, baseline_profile_info={}, diff=diff, changed=True)
+            self.module.exit_json(msg=SUCCESS_DELETION_MSG, baseline_profile_info={}, changed=True)
+        else:
+            self.module.exit_json(msg=FAILED_DELETION_MSG, failed=True)
+
+    def execute(self):
+        vcenter_uuid = self.module.params.get('vcenter_uuid')
+        profile_exists = self.omevv_baseline_obj.get_baseline_profile_by_name(self.profile_name, vcenter_uuid)
+
+        if profile_exists and self.module.check_mode and self.module._diff:
+            diff = self.diff_mode_check(profile_exists)
+            self.module.exit_json(msg=CHANGES_FOUND_MSG, diff=diff, changed=True)
+        if not profile_exists and self.module.check_mode:
+            self.module.exit_json(msg=CHANGES_NOT_FOUND_MSG, changed=False)
+        if not profile_exists and not self.module.check_mode and self.module._diff:
+            self.module.exit_json(msg=CHANGES_NOT_FOUND_MSG, diff={"before": {}, "after": {}}, changed=False)
+        if not profile_exists and not self.module.check_mode:
+            self.module.exit_json(msg=CHANGES_NOT_FOUND_MSG, changed=False)
+        if profile_exists and not self.module.check_mode:
+            self.perform_delete_baseline_profile(profile_exists)
+        if profile_exists and self.module.check_mode:
+            self.module.exit_json(msg=CHANGES_FOUND_MSG, changed=True)
+
+
+def handle_http_error(err, module):
```

Comment:
is it possible if we can move this to common omevv utils file or in omevv.py ? or try to generalize it so we can use the same in future module, same for other functions which is handline errors

Reply:
handled for my module specific